Xamarin.PropertyEditing.Mac/Controls/Custom/NumericSpinEditor.cs

```diff
@@ -0,0 +1,287 @@
+﻿using System;
+using AppKit;
+using CoreGraphics;
+using Foundation;
+
+namespace Xamarin.PropertyEditing.Mac
+{
+	public class NumericSpinEditor : NSView
+	{
+		protected NumericTextField numericEditor;
+		protected NSStepper stepper;
+		protected bool editing;
+
+
+		public event EventHandler ValueChanged;
+		public event EventHandler StepperValueChanged;
+		public event EventHandler FieldEditingEnded;
+
+		public ValidationType NumericMode {
+			get { return numericEditor.NumericMode; }
+			set {
+				numericEditor.NumericMode = value;
+				Reset ();
+			}
+		}
+
+		public string StringValue {
+			get { return numericEditor.StringValue; }
+			set { numericEditor.StringValue = value; }
+		}
+
+		public string PlaceholderString {
+			get { return ((NSTextFieldCell)numericEditor.Cell).PlaceholderString; }
+			set { ((NSTextFieldCell)numericEditor.Cell).PlaceholderString = value; }
+		}
+
+		public NSStepper Stepper {
+			get { return stepper; }
+		}
+
+		public override CGSize IntrinsicContentSize {
+			get {
+				var baseSize = stepper.IntrinsicContentSize;
+				return new CGSize (baseSize.Width + 20, baseSize.Height);
+			}
+		}
+
+		public NSColor BackgroundColor {
+			get {
+				return numericEditor.BackgroundColor;
+			}
+			set {
+				numericEditor.BackgroundColor = value;
+			}
+		}
+
+		public override nfloat BaselineOffsetFromBottom {
+			get { return numericEditor.BaselineOffsetFromBottom; }
+		}
+
+		public int Digits {
+			get { return (int)formatter.MaximumFractionDigits; }
+			set { formatter.MaximumFractionDigits = value; }
+		}
+
+		public virtual double Value {
+			get { return stepper.DoubleValue; }
```

Comment:
I think we should not have multiple public ways of setting values when each way does something different. This control is literally impossible for someone to use with the way it's currently written. These kinds of bugs have been the bane of my life for the last few months :( See my main comment at the end. There's definitely got to be a better way of handling this other than trying to publicly expose the raw string value and double value :)